packages/backend/src/modules/activity2/indexers/ActivityIndexer.ts

```diff
@@ -0,0 +1,69 @@
+import { ActivityRepository } from '@l2beat/database/src/activity/repository'
+import { ProjectId, UnixTime } from '@l2beat/shared-pure'
+import {
+  ManagedChildIndexer,
+  ManagedChildIndexerOptions,
+} from '../../../tools/uif/ManagedChildIndexer'
+import { TxsCountProvider } from '../services/TxsCountProvider'
+
+export interface ActivityIndexerDeps
+  extends Omit<ManagedChildIndexerOptions, 'name'> {
+  projectId: ProjectId
+  txsCountProvider: TxsCountProvider
+  activityRepository: ActivityRepository
+  batchSize?: number
+}
+
+export class ActivityIndexer extends ManagedChildIndexer {
+  constructor(private readonly $: ActivityIndexerDeps) {
+    const logger = $.logger.tag($.projectId)
+    super({ ...$, logger, name: `activity_indexer_${$.projectId}` })
+  }
+
+  override async update(from: number, to: number): Promise<number> {
+    const batchSize = this.$.batchSize ?? 100
```

Comment:
batch size should always be defined, this ?? will be used during object instantiation